diff --git a/MiniTwitch.Helix/Responses/EventSubSubscriptions.cs b/MiniTwitch.Helix/Responses/EventSubSubscriptions.cs
index c9ed3c0..53e8a0d 100644
--- a/MiniTwitch.Helix/Responses/EventSubSubscriptions.cs
+++ b/MiniTwitch.Helix/Responses/EventSubSubscriptions.cs
@@ -1,4 +1,5 @@
 using System.Text.Json.Serialization;
+using MiniTwitch.Helix.Enums;
 using MiniTwitch.Helix.Models;
 
 namespace MiniTwitch.Helix.Responses;
@@ -19,7 +20,8 @@ public class EventSubSubscriptions : PaginableResponse<EventSubSubscriptions.Sub
 
     public record Subscription(
         string Id,
-        string Status,
+        [property: JsonConverter(typeof(JsonStringEnumConverter))]
+        EventSubStatus Status,
         string Type,
         string Version,
         Condition Condition,
